### ISSUE TITLE
[Accessibility]: Expanding search matches should focus the first previously-hidden match, not the collapse button

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#43143]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34190

### Problem description

For content match search results, expanding the "Show $x more matches" keeps focus on that button (which now reads "Show less". This is a problem because keyboard users have to tab back and navigate the new matches in reverse order.

<img width="874" alt="image" src="https://user-images.githubusercontent.com/206864/196559540-b3b2193f-26cf-4cf7-8f01-932f5ec48e26.png">

### Expected behavior

The first previously-hidden match should be focus on expansion.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

/cc @lguychard
